demos/sample_azure_iot_adu/sample_azure_iot_adu.c

```diff
@@ -918,33 +888,72 @@ static void prvAzureDemoTask( void * pvParameters )
+                                                                 sampleazureiotPROCESS_LOOP_TIMEOUT_MS );
+                        configASSERT( xResult == eAzureIoTSuccess );
+
+                        /* In prvDownloadUpdateImageIntoFlash, we make a call to the _ProcessLoop() function */
+                        /* which could bring in a new or cancelled update. */
+                        /* Check xProcessUpdateRequest and the action again in case a new version came in that was invalid. */
+                        if( xProcessUpdateRequest && ( xAzureIoTAduUpdateRequest.xWorkflow.xAction == eAzureIoTADUActionApplyDownload ) )
+                        {
+                            xResult = prvEnableImageAndResetDevice();
+                            configASSERT( xResult == eAzureIoTSuccess );
+
+                            xResult = prvSpoofNewVersion();
+                            configASSERT( xResult == eAzureIoTSuccess );
+                        }
+                        else
+                        {
+                            xResult = AzureIoTADUClient_SendAgentState( &xAzureIoTADUClient,
+                                                                        &xAzureIoTHubClient,
+                                                                        &xADUDeviceProperties,
+                                                                        &xAzureIoTAduUpdateRequest,
+                                                                        eAzureIoTADUAgentStateIdle,
+                                                                        NULL,
+                                                                        ucScratchBuffer,
+                                                                        sizeof( ucScratchBuffer ),
+                                                                        NULL );
+
+                            xProcessUpdateRequest = false;
+                        }
+                    }
+                    else
+                    {
+                        LogInfo( ( "Unknown action received: %i", xAzureIoTAduUpdateRequest.xWorkflow.xAction ) );
+                    }
                 }
-                else
-                {
-                    LogInfo( ( "Unknown action received: %i", xAzureIoTAduUpdateRequest.xWorkflow.xAction ) );
-                }
-            }
 
-            /* Leave Connection Idle for some time. */
-            LogInfo( ( "Keeping Connection Idle..." ) );
-            vTaskDelay( sampleazureiotDELAY_BETWEEN_PUBLISHES_TICKS );
-        }
+                /* Leave Connection Idle for some time. */
+                LogInfo( ( "Keeping Connection Idle..." ) );
+                vTaskDelay( sampleazureiotDELAY_BETWEEN_PUBLISHES_TICKS );
+            }
 
-        xResult = AzureIoTHubClient_UnsubscribeProperties( &xAzureIoTHubClient );
-        configASSERT( xResult == eAzureIoTSuccess );
+            if( xAzureSample_IsConnectedToInternet() )
```

Comment:
Also, should this be outside of the for( ; xAzureSample_IsConnectedToInternet(); ) so that everything gets shut down properly if the connection is dropped?